cs/test/NativeReadCacheTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Linq;
+using FASTER.core;
+using System.IO;
+using NUnit.Framework;
+
+namespace FASTER.test
+{
+
+    [TestFixture]
+    internal class NativeReadCacheTests
+    {
+        private FasterKV<KeyStruct, ValueStruct, InputStruct, OutputStruct, Empty, Functions> fht;
+        private IDevice log;
+
+        [SetUp]
+        public void Setup()
+        {
+            log = Devices.CreateLogDevice(TestContext.CurrentContext.TestDirectory + "\\hybridlog_native.log", deleteOnClose: true);
+            fht = new FasterKV<KeyStruct, ValueStruct, InputStruct, OutputStruct, Empty, Functions>
+                (1L<<20, new Functions(), new LogSettings { LogDevice = log, MemorySizeBits = 15, PageSizeBits = 10, ReadCacheSizeBits = 10 });
+            fht.StartSession();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            fht.StopSession();
+            fht.Dispose();
+            fht = null;
+            log.Close();
+        }
+
+        [Test]
+        public void NativeDiskWriteRead()
+        {
+            InputStruct input = default(InputStruct);
+
+            Random r = new Random(10);
+
+            for (int i = 0; i < 2000; i++)
+            {
+                var key1 = new KeyStruct { kfield1 = i, kfield2 = i + 1 };
+                var value = new ValueStruct { vfield1 = i, vfield2 = i + 1 };
+                fht.Upsert(ref key1, ref value, Empty.Default, 0);
+            }
+            fht.CompletePending(true);
```

Comment:
Perhaps you should ShiftHeadAddress(LogTailAddress), not only to ensure the synchronicity of the reads is actually due to the read cache but also to reflect the main use case, where the hybrid log in-memory regions will be unnecessary after data load

Reply:
Expanded the test to cover this.